src/main/java/com/contentful/java/cda/CDAHttpException.java

```diff
@@ -37,6 +40,9 @@ public CDAHttpException(Request request, Response response) {
 
   private String readResponseBody(Response response) {
     try {
+      BufferedSource bufferedSource = response.body().source();
+      Timeout timeout = bufferedSource.timeout();
+      timeout.deadline(1, TimeUnit.SECONDS);
```

Comment:
Before the tests needed for ever, due to the fact that it tried for forever to get more out of the buffered source. Since everything should be downloaded already, setting it to one second should be fine ... ;)